src/basic/user-util.c

```diff
@@ -465,6 +465,54 @@ int in_group(const char *name) {
         return in_gid(gid);
 }
 
+static
+bool gid_list_has(const gid_t *list1, int size1, gid_t val) {
+        for (int i = 0; i < size1; i++)
+                if (list1[i] == val)
+                        return true;
+        return false;
+}
+
+int merge_gid_lists(const gid_t *list1, int size1, const gid_t *list2, int size2, gid_t **ret) {
```

Comment:
int → size_t

Reply:
(and yes, it's OK to return the resulting size in an `int` return value, i.e. be asymmetric with the types for input and output here. For the output you have a reason: you want to return an error was well, but for the input, please use the most accurate type, and that's size-t)